src/storage/secondary/rowset/rowset_iterator.rs

```diff
@@ -52,16 +49,24 @@ impl RowSetIterator {
             panic!("more than 1 row handler column")
         }
 
-        if row_handler_count == column_refs.len() {
-            panic!("no user column")
-        }
-
         let mut column_iterators: Vec<Option<ColumnIteratorImpl>> = vec![];
 
         for column_ref in &*column_refs {
             // TODO: parallel seek
             match column_ref {
-                StorageColumnRef::RowHandler => column_iterators.push(None),
+                StorageColumnRef::RowHandler => {
+                    let column = rowset.column(0);
+                    let row_count = column
+                        .index()
+                        .indexes()
+                        .iter()
+                        .fold(0, |acc, index| acc + index.row_count);
+                    column_iterators.push(Some(ColumnIteratorImpl::new_row_handler(
```

Comment:
`column_iterators` doesn't need to store `Option`. It can be `ColumnIteratorImpl`. You may do the refactor later.